lib/ghomperm.gi

```diff
@@ -1621,15 +1623,46 @@ InstallGlobalFunction( ImageKernelBlocksHomomorphism, function( hom, H,par )
 
             # Make <S> the stabilizer of the block <B>.
             InsertTrivialStabilizer( S.stabilizer, B[ 1 ] );
-            j := 1;
-            while                                j < Length( B )
-                  and Length( S.stabilizer.orbit ) < Length( B )  do
-                j := j + 1;
-                if IsBound( S.translabels[ B[ j ] ] )  then
-                    AddGeneratorsExtendSchreierTree( S.stabilizer,
-                            [ InverseRepresentative( S, B[ j ] ) ] );
-                fi;
-            od;
+
+            if Length(B)>Length(D)^2 then
+              # if there are few, large blocks the search through all block
+              # points is tedious. Rather use an orbit/stabilizer algorithm.
+              orb:=[i];
+              rep:=[One(H)];
+              j:=1;
+              while j<=Length(orb) do
+                for k in S.generators do
+                  img:=D[orb[j]][1]^k;
+		  p:=hom!.reps[img];
```

Comment:
Could you please avoid using tabs for indentation in this line?

Reply:
A hello from my editor. I'll fix it.

Reply:
PS: Is there a problem if I simply replace *all* tabs?

Reply:
@hulpke I would recommend to replace ALL tabs only in separate commits (not necessarily in this PR). Otherwise, diffs will be very lengthy and it will be VERY hard to review this PR.

Reply:
@alex-konovalov 
The tab changes make up less than 1/3 of the overall PR, but I kept it a separate commit so it can be thrown out again.